Sources/SWBCore/Settings/Settings.swift

```diff
@@ -2436,6 +2436,13 @@ private class SettingsBuilder {
             platformTable.push(macro, realArchs)
         }
 
+        // Add ObjC ARC support for Apple platforms.
+        if core.hostOperatingSystem == .macOS {
```

Comment:
Thanks for moving this into the platform settings! This check should use the targeted platform instead of the host platform, since the build system may be cross-compiling from macOS to a non-Apple platform. We also want to ensure this applies when the destination is iOS/watchOS/tvOS/visionOS

Reply:
Sure.